### DISCUSSION TITLE
How to publish properties and tags for TODOs in post sub-headings

### DISCUSSION THREAD

**Discussion by stefanv** [kaushalmodi/ox-hugo#512] [Q&A] [answered]
I use ox-roam to export my collection of org-roam notes ("one post per file" mode).  I've noticed that TODO properties (such as DEADLINE and SCHEDULED) and tags do not survive publishing.  Is there a way to get them to appear in the output?

**Comment by kaushalmodi** [comment]
Can you post an Org example and the expected markdown?

**Reply by stefanv** [reply]
```org
* TODO Contribute back to ox-hugo                            :@stefanv:
SCHEDULED: <2022-01-20 Mon> DEADLINE: <2022-01-30 Mon>
```

Output maybe something like:

```
## <span class="org-todo todo TODO">TODO</span> Contribute back to ox-hugo {#contribute-back-to-ox-hugo} <div class="tag">@stefanv</div><div class="property-name">SCHEDULED</div><div class="property-content date">2022-01-20 Mon</div><div class="property-name">DEADLINE</div><div class="property-name date">2022-01-30 Mon</div>
```

**Reply by kaushalmodi** [reply]
Is "Contribute back to ox-hugo" the heading of a post or a sub-heading within a post?

**Reply by stefanv** [reply]
That is a sub-heading within a post.

**Reply by stefanv** [reply]
I have documents that list tasks inside of them. I.e., I don't make blog posts, but I publish entire documents that happen to have TODO headings in them, just like one would normally use org.

**Reply by kaushalmodi** [reply]
If you can find a way to publish span tags for headings in Markdown, we can make this work.

Note that div tags inside headings are illegal, but we can look into exporting the dates as part of the heading in span tags.

But I am not finding any example on how to do that.

**Reply by kaushalmodi** [reply]
I tried out an HTML export and I can see the tags being exported in span tags if I set the `tags:t` export setting. But the DEADLINE/SCHEDULED dates never get exported.

**Reply by stefanv** [reply]
It looks like this comes out when you use `org-export-with-planning` only.

**Reply by kaushalmodi** [reply]
Thanks. Note to self: `#+options: tags:t p:t`

**Reply by kaushalmodi** [reply]
ok, so this Org:

```org
#+options: tags:t p:t

* TODO Contribute back to ox-hugo                          :@stefanv:abc:def:
SCHEDULED: <2022-01-20 Mon> DEADLINE: <2022-01-30 Mon>
```

exports to this HTML:

```html
<div id="outline-container-org8e9585e" class="outline-2">
  <h2 id="org8e9585e">
    <span class="section-number-2">1.
    </span>
    <span class="todo TODO">TODO
    </span> Contribute back to ox-hugo&#xa0;&#xa0;&#xa0;
    <span class="tag">
      <span class="_stefanv">@stefanv
      </span>&#xa0;
      <span class="abc">abc
      </span>&#xa0;
      <span class="def">def
      </span>
    </span>
  </h2>
  <div class="outline-text-2" id="text-1">
    <p>
      <span class="timestamp-wrapper">
        <span class="timestamp-kwd">DEADLINE:
        </span>
        <span class="timestamp">&lt;2022-01-30 Sun&gt;
        </span>
        <span class="timestamp-kwd">SCHEDULED:
        </span>
        <span class="timestamp">&lt;2022-01-20 Thu&gt;
        </span>
      </span>
    </p>
  </div>
</div>
```

What if the ox-hugo export tweaks this slightly and moves the export of tags outside the heading as well.. into a separate span (or div) tag?

**Reply by stefanv** [reply]
That would be great! Then I can just style them the way I want in CSS.

I now have the following in my publishing conf:

```elisp
(setq org-export-with-tags t)
(setq org-export-with-planning t)
```

I now see deadline/schedule in the markdown output, but still no tags.  I guess this is specific to ox-hugo, because it does appear in HTML and standard markdown.

**Reply by kaushalmodi** [reply]
> I guess this is specific to ox-hugo

Yes, I need to enable exporting of them.

**Reply by kaushalmodi** [reply]
> If you can find a way to publish span tags for headings in Markdown, we can make this work.

I just recalled that when I added support for TODO classes in headings, I was able to export `span` tags as-is. So I should be able to export the tags in span tags as well.

**Comment by kaushalmodi** [comment, marked as answer]
@stefanv 

Now these options work: `#+options: tags:t p:t`

- [x] Export planning info -- https://github.com/kaushalmodi/ox-hugo/pull/515
- [x] Export sub-heading tags -- https://github.com/kaushalmodi/ox-hugo/pull/516
